packages/netlify-cms-core/src/components/Collection/Entries/Entries.js

```diff
@@ -13,34 +24,40 @@ const Entries = ({
   cursor,
   handleCursorActions,
   t,
+  page,
 }) => {
   const loadingMessages = [
     t('collection.entries.loadingEntries'),
     t('collection.entries.cachingEntries'),
     t('collection.entries.longerLoading'),
   ];
 
-  if (entries) {
-    return (
-      <EntryListing
-        collections={collections}
-        entries={entries}
-        viewStyle={viewStyle}
-        cursor={cursor}
-        handleCursorActions={handleCursorActions}
-      />
-    );
+  if (isFetching && page === undefined) {
+    return <Loader active>{loadingMessages}</Loader>;
   }
 
-  if (isFetching) {
-    return <Loader active>{loadingMessages}</Loader>;
+  if (entries && entries.size > 0) {
+    return (
+      <>
+        <EntryListing
+          collections={collections}
+          entries={entries}
+          viewStyle={viewStyle}
+          cursor={cursor}
+          handleCursorActions={handleCursorActions}
+        />
+        {isFetching && page !== undefined ? (
+          <PaginationMessage>{t('collection.entries.loadingEntries')}</PaginationMessage>
+        ) : null}
+      </>
+    );
   }
 
-  return <div className="nc-collectionPage-noEntries">No Entries</div>;
+  return <NoEntriesMessage>{t('collection.entries.noEntries')}</NoEntriesMessage>;
```

Comment:
This code was never reached as well. Now if you don't have entries you'll see it in the UI

Reply:
Oh, great catch!